arangod/RocksDBEngine/RocksDBDumpContext.cpp

```diff
@@ -452,28 +452,34 @@ void RocksDBDumpContext::handleWorkItem(WorkItem item) {
       continue;
     }
 
-    auto storedSlice = [&]() -> VPackSlice {
+    // Sanitize document
+    VPackValueLength const inputLength = documentSlice.byteSize();
+    VPackBuffer<uint8_t> tmpBuffer;
+    bool resolveExt =
+        basics::VelocyPackHelper::hasNonClientTypes(documentSlice);
+    if (resolveExt) {                  // resolve
+      tmpBuffer.reserve(inputLength);  // reserve space already
```

Comment:
The sanitation will usually produce slightly more bytes, since one-byte abbreviations will be replaced by strings like "_key" which needs 5 bytes. Furthermore, the custom type for the `_id` attribute will be replaced by the full string. Therefore, I suggest to reserve a bit more here, let's say `inputLength + 64` to avoid a double allocation. If the key is very long, we still run into a double allocation, but this can be tolerated. I do not think it is worthwhile to determine the length of the collection name and the key and accurately compute the needed reserve.